File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/model/Room.java
package org.learn.learningmanagementbackend.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import org.learn.learningmanagementbackend.enums.RoomType;

import java.util.List;

@Getter
@Setter
@Entity
@NoArgsConstructor
@Table(name = "Room")
public class Room extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", referencedColumnName = "id")
    private SchoolBranch schoolBranch;

    @Column(name = "building", length = 50)
    private String building;

    @Column(name = "room_number", length = 20)
    private String roomNumber;

    @Column(name = "capacity")
    private Integer capacity;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    private RoomType roomType;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "equipment", columnDefinition = "json")
    private List<String> equipment;

    @Column(name = "is_active")
    private Boolean isActive;
}

File: learning_management_app/lib/models/admin/room.dart
class Room {
  final int? id;
  final int? schoolBranchId;
  final String? schoolBranchName;
  final String? roomNumber;
  final String? building;
  final String? roomType;
  final int? capacity;
  final List<String>? equipment;
  final bool? isActive;

  Room({
    this.id,
    this.schoolBranchId,
    this.schoolBranchName,
    this.roomNumber,
    this.building,
    this.roomType,
    this.capacity,
    this.equipment,
    this.isActive,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'],
      schoolBranchId: json['schoolBranchId'],
      schoolBranchName: json['schoolBranchName'],
      roomNumber: json['roomNumber'],
      building: json['building'],
      roomType: json['roomType'],
      capacity: json['capacity'],
      equipment: json['equipment'] != null
          ? List<String>.from(json['equipment'])
          : null,
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'schoolBranchId': schoolBranchId,
      'schoolBranchName': schoolBranchName,
      'roomNumber': roomNumber,
      'building': building,
      'roomType': roomType,
      'capacity': capacity,
      'equipment': equipment,
      'isActive': isActive,
    };
  }
}
